ql/math/randomnumbers/xoshiro256starstaruniformrng.cpp

```diff
@@ -0,0 +1,41 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ Copyright (C) 2023 Ralf Konrad Eckel
+
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+#include <ql/math/randomnumbers/seedgenerator.hpp>
+#include <ql/math/randomnumbers/splitmix64.hpp>
+#include <ql/math/randomnumbers/xoshiro256starstaruniformrng.hpp>
+
+namespace QuantLib {
+
+    Xoshiro256StarStarUniformRng::Xoshiro256StarStarUniformRng(uint64_t seed) {
+        SplitMix64 splitMix64(seed != 0 ? seed : SeedGenerator::instance().get());
+        do {
+            s0_ = splitMix64.next();
+            s1_ = splitMix64.next();
+            s2_ = splitMix64.next();
+            s3_ = splitMix64.next();
+        } while (s0_ == 0 && s1_ == 0 && s2_ == 0 && s3_ == 0);
```

Comment:
I don't think this is ever going to happen.

Reply:
I fully agree, it's very, very unlikely.

I have "stolen" the idea from `c#`. Random number generator guys are probably extra ~paranoid~ careful. 😄

Personally I would keep it here as a reminder what theoretically might go wrong. But I leave it up to you.

https://github.com/dotnet/runtime/blob/a5ff66c825fab95898c84a57ff305bece82f352d/src/libraries/System.Private.CoreLib/src/System/Random.Xoshiro256StarStarImpl.cs#L33-L45

Reply:
I'd take it out.  If the splitmix rng gives you 4 zeroes in a row, it's probably going to keep giving you zeroes.